cddl_gen/cddl_gen.py

```diff
@@ -201,15 +204,16 @@ def from_cddl(cddl_class, cddl_string, default_max_qty, *args, **kwargs):
         # Parse the definitions, replacing the each string with a
         # CodeGenerator instance.
         for my_type, cddl_string in type_strings.items():
-            parsed = cddl_class(*args, default_max_qty, my_types, **kwargs, base_name=my_type)
+            parsed = cddl_class(*args, default_max_qty, my_types, **kwargs,
+                                base_name=my_type, base_stem=my_type)
             parsed.get_value(cddl_string.replace("\n", " "))
             my_types[my_type] = parsed.flatten()[0]
-            my_types[my_type].set_id_prefix("")
 
         counter(True)
 
         # post_validate all the definitions.
         for my_type in my_types:
+            my_types[my_type].set_id_prefix("")
```

Comment:
Couldn't this be done in a class initialiser or something? Setting a non-value (which I consider the empty string) indicates that there is a flaw somewhere IMHO. Also the comment above should be updated to reflect what is done here.

Reply:
Or we could set the default value in the function definition.

Reply:
>  Setting a non-value (which I consider the empty string) indicates that there is a flaw somewhere 

I have to disagree when it comes to Python. Both `''` and `None` have very different cases in Python contexts, since `None` is `NoneType` but `''` is `str`. If your function expects an argument to be a string, even an empty one, then `''` is always preferable to `None`.

Often times `None` can be a tricky and problematic value to pass around since it has logical equivalency to both `''` and `0` with incompatible functionality in use.

Reply:
This for loop iterates over all the top-level types. These can have sub-types, and sub-sub-types etc. (e.g. if they are lists). So this function sets an "initial value" which is appended to, and passed to sub-types. This means it needs the structure to already exists, so it needs to happen afterwards. This was moved because `set_id_prefix()` was changed to now use a function that needs even more things to happen before it can be called.